driver-core/src/main/com/mongodb/client/model/expressions/DocumentExpression.java

```diff
@@ -28,54 +28,65 @@
  */
 public interface DocumentExpression extends Expression {
 
-    <R extends DocumentExpression> R setField(String path, Expression exp);
+    DocumentExpression setField(String path, Expression exp);
 
     DocumentExpression unsetField(String path);
 
 
     BooleanExpression getBoolean(String field);
 
-    BooleanExpression getBoolean(String field, BooleanExpression orElse);
+    BooleanExpression getBoolean(String field, BooleanExpression other);
 
-    default BooleanExpression getBoolean(final String field, final boolean orElse) {
-        return getBoolean(field, of(orElse));
+    default BooleanExpression getBoolean(final String field, final boolean other) {
+        return getBoolean(field, of(other));
     }
 
     NumberExpression getNumber(String field);
 
-    NumberExpression getNumber(String field, NumberExpression orElse);
+    NumberExpression getNumber(String field, NumberExpression other);
 
-    default NumberExpression getNumber(final String field, final Number orElse) {
-        return getNumber(field, Expressions.numberToExpression(orElse));
+    default NumberExpression getNumber(final String field, final Number other) {
+        return getNumber(field, Expressions.numberToExpression(other));
     }
 
     IntegerExpression getInteger(String field);
 
+    IntegerExpression getInteger(String field, IntegerExpression other);
+
+    default NumberExpression getInteger(final String field, final int other) {
+        return getInteger(field, (IntegerExpression) Expressions.numberToExpression(other));
+    }
+
+    default NumberExpression getInteger(final String field, final long other) {
+        return getInteger(field, (IntegerExpression) Expressions.numberToExpression(other));
+    }
```

Comment:
Two notes here:

1. These methods should return `IntegerExpression`.
2. Why is casting `(IntegerExpression)` used instead of implementing the methods as `return getInteger(field, Expressions.of(other));`?